pkg/feature/manifest.go

```diff
@@ -59,57 +208,99 @@ func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	return manifests, nil
 }
 
-func createManifestFrom(fsys fs.FS, path string) manifest {
+func CreateBaseManifestFrom(fsys fs.FS, path string) *baseManifest { //nolint:golint,revive //No need to export baseManifest.
 	basePath := filepath.Base(path)
-	m := manifest{
-		name:     basePath,
-		path:     path,
-		patch:    strings.Contains(basePath, ".patch"),
-		template: filepath.Ext(path) == ".tmpl",
-		fsys:     fsys,
+	m := &baseManifest{
+		name:  basePath,
+		path:  path,
+		patch: strings.Contains(basePath, ".patch"),
+		fsys:  fsys,
 	}
 
 	return m
 }
 
-func (m *manifest) targetPath() string {
-	return fmt.Sprintf("%s%s", m.path[:len(m.path)-len(filepath.Ext(m.path))], ".yaml")
+func CreateTemplateManifestFrom(fsys fs.FS, path string) *templateManifest { //nolint:golint,revive //No need to export templateManifest.
+	basePath := filepath.Base(path)
+	m := &templateManifest{
+		name:  basePath,
+		path:  path,
+		patch: strings.Contains(basePath, ".patch"),
+		fsys:  fsys,
+	}
+
+	return m
 }
 
-func (m *manifest) process(data interface{}) error {
-	manifestFile, err := m.open()
-	if err != nil {
-		return err
+func CreateKustomizeManifestFrom(path string, fsys filesys.FileSystem) *kustomizeManifest { //nolint:golint,revive //No need to export kustomizeManifest.
+	basePath := filepath.Base(path)
+	if basePath == KustomizationFile {
+		path = filepath.Dir(path)
 	}
-	defer manifestFile.Close()
-
-	content, err := io.ReadAll(manifestFile)
-	if err != nil {
-		return fmt.Errorf("failed to create file: %w", err)
+	if fsys == nil {
+		fsys = filesys.MakeFsOnDisk()
```

Comment:
What's the reason for the fall back here? Maybe we can just pass this struct instead when we call it?

Reply:
[6b72743](https://github.com/opendatahub-io/opendatahub-operator/pull/867/commits/6b727430b5c728fa77c8300bbe748bdd847405be) done here

Reply:
I am not sure we were thinking about the same thing.  I was wondering why we have

```
if fsys == nil {
		fsys = filesys.MakeFsOnDisk()
}
```

if we can just pass it. :) But the other one seemed unnecessary too.